openshift-knative-operator/pkg/serving/extension_test.go

```diff
@@ -121,10 +142,147 @@ func TestReconcile(t *testing.T) {
 			if objs == nil {
 				objs = []runtime.Object{defaultIngress}
 			}
+			ks := c.in.DeepCopy()
 			ctx, _ := ocpfake.With(context.Background(), objs...)
+			ctx, _ = kubefake.With(ctx, &servingNamespace)
+			ext := NewExtension(ctx)
+			ext.Reconcile(context.Background(), ks)
+			// Ignore time differences.
+			opt := cmp.Comparer(func(apis.VolatileTime, apis.VolatileTime) bool {
+				return true
+			})
+			if !cmp.Equal(ks, c.expected, opt) {
+				t.Errorf("Got = %v, want: %v, diff:\n%s", ks, c.expected, cmp.Diff(ks, c.expected, opt))
+			}
+		})
+	}
+}
 
+func TestMonitoring(t *testing.T) {
+	cases := []struct {
+		name     string
+		in       *v1alpha1.KnativeServing
+		objs     []runtime.Object
+		expected *v1alpha1.KnativeServing
+		// Returns the expected status for monitoring
+		setupMonitoringToggle func() (bool, error)
```

Comment:
I don't quite parse why this is a function :thinking:. You seem to be returning just a boolean everytime anyway. Any reason why an error is used here?

Reply:
Error comes from the the os.SetEnv call.

Reply:
But you are, i think?

Reply:
Yes because `toggleEnvVar` should return error not false, I will fix that.